zipkin-local.js

```
@@ -0,0 +1,129 @@
+    return;
+  }
+  if (logger.queue.length > 0) {
+    const formattedQueue = logger.queue.map((trace) => {
+      const formatTrace = JSON.parse(trace);
+      formatTrace.localEndpoint = {};
+      formatTrace.localEndpoint.serviceName = THEME_OPTIONS.SITE_NAME;
+      formatTrace.tags = {};
+      if (formatTrace.annotations) {
+        delete formatTrace["annotations"];
+      }
+      if (formatTrace.binaryAnnotations) {
+        for (let anno of formatTrace.binaryAnnotations) {
+          formatTrace.tags[anno.key] = anno.value;
+        }
+        delete formatTrace["binaryAnnotations"];
+      }
+      if (formatTrace.name === "run-api") {
+        formatTrace.name += `: ${formatTrace.tags.api}`;
+      }
+      if (formatTrace.name === "run-plugin") {
+        formatTrace.name += `: ${formatTrace.tags.plugin}`;
+      }
+      const { tags } = THEME_OPTIONS.traces;
+      for (let tag in tags) {
+        formatTrace.tags[tag] = tags[tag];
+      }
+      formatTrace.tags.buildId = THEME_OPTIONS.buildId;
+      formatTrace.tags.gatsbySite = THEME_OPTIONS.SITE_NAME;
+      return JSON.stringify({
+        ...formatTrace,
+        ...THEME_OPTIONS.traces.tags,
+      });
+    });
+
+    const postBody = `[${formattedQueue.join(",")}]`;
+
+    try {
+      const response = await (0, _nodeFetch.default)(logger.endpoint, {
+        method: `POST`,
+        body: postBody,
+        headers: {
+          "Content-Type": "application/json",
+          "Api-Key": THEME_OPTIONS.NR_INGEST_KEY,
+          "Data-Format": "zipkin",
+          "Data-Format-Version": 2,
+        },
+      });
+
+      if (response.status !== 202) {
+        const err =
+          `[@] gatsby-plugin-newrelic: Unexpected response while sending trace data, status:` +
+          `${response.status}, body: ${postBody}`;
+        if (logger.errorListenerSet) logger.emit(`error`, new Error(err));
+        else console.error(err);
```

If / else statements without braces (`{ }`) can very easily introduce bugs. I'd suggest adding those in:

```suggestion
        if (logger.errorListenerSet) {
          logger.emit(`error`, new Error(err));
        } else {
          console.error(err);
        }
```

---

Furthermore, we have a `catch` block that handles errors down below. Perhaps we can just throw the error up here and let the block down below handle communicating errors.

---

again this is from the original compilation from Typescript so i'll definitely test this (need some shorthand for this 😂 

OG Typescript: will test 🤠